### PR TITLE
fix: use @eslint/create-config latest

### PR DESCRIPTION
**Pull request by aladdin-add**
refs: https://github.com/eslint/eslint/issues/18369, https://github.com/eslint/eslint/discussions/18361,

#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)
We've received a few reports generated configs seem to be eslintrc format, so I've updated the documentation to use the latest version.

### CODE REVIEW

File: bin/eslint.js
Comment (fasttime) [inline bin/eslint.js:1]: Shouldn't the `console.warn` on line 143 suggest to run `npm init @eslint/config@latest`?

**Review by snitin315 — APPROVED** [review]
LGTM, thanks!

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks! I'm fine with this solution.

**Comment by aladdin-add** [comment]
Technically, it's not an eslint's issue; but it does appear that some users have encountered, so I'm 👍 .

**Review by fasttime — APPROVED** [review]
LGTM, thanks!

**Pull request merged**